src/Nethermind/Nethermind.Evm/Tracing/GethStyle/GethLikeTxTraceCollection.cs

```diff
@@ -1,18 +1,25 @@
 // SPDX-FileCopyrightText: 2025 Demerzel Solutions Limited
 // SPDX-License-Identifier: LGPL-3.0-only
 
+using System;
 using System.Collections;
 using System.Collections.Generic;
 using System.Text.Json.Serialization;
+using Nethermind.Core.Extensions;
 
 namespace Nethermind.Evm.Tracing.GethStyle;
 
 [JsonConverter(typeof(GethLikeTxTraceCollectionConverter))]
-public record GethLikeTxTraceCollection(IReadOnlyCollection<GethLikeTxTrace> Traces) : IReadOnlyCollection<GethLikeTxTrace>
+public record GethLikeTxTraceCollection(IReadOnlyCollection<GethLikeTxTrace> Traces) : IReadOnlyCollection<GethLikeTxTrace>, IDisposable
 {
     public IEnumerator<GethLikeTxTrace> GetEnumerator() => Traces.GetEnumerator();
 
     IEnumerator IEnumerable.GetEnumerator() => GetEnumerator();
 
     public int Count => Traces.Count;
+    public void Dispose()
```

Comment:
This feels wrong somehow. Why is a record that seems to be used only for JSON serialization need to be responsible for disposing stuff?

Reply:
Yes we changed our JSON serialization to allow IDisposables to be serialized and then disposing it. It allows using pooled memory when responding to JSON requests instead of allocation. And I don't have better ideas on how to do it.